clients/go/admin/config_flags.go

```diff
@@ -50,11 +41,11 @@ func (Config) mustMarshalJSON(v json.Marshaler) string {
 // flags is json-name.json-sub-name... etc.
 func (cfg Config) GetPFlagSet(prefix string) *pflag.FlagSet {
 	cmdFlags := pflag.NewFlagSet("Config", pflag.ExitOnError)
-	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "endpoint"), defaultConfig.Endpoint.String(), "For admin types,  specify where the uri of the service is located.")
-	cmdFlags.Bool(fmt.Sprintf("%v%v", prefix, "insecure"), defaultConfig.UseInsecureConnection, "Use insecure connection.")
-	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "maxBackoffDelay"), defaultConfig.MaxBackoffDelay.String(), "Max delay for grpc backoff")
-	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "perRetryTimeout"), defaultConfig.PerRetryTimeout.String(), "gRPC per retry timeout")
-	cmdFlags.Int(fmt.Sprintf("%v%v", prefix, "maxRetries"), defaultConfig.MaxRetries, "Max number of gRPC retries")
+	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "ClientBaseConfig.endpoint"), defaultConfig.ClientBaseConfig.Endpoint.String(), "For admin types,  specify where the uri of the service is located.")
+	cmdFlags.Bool(fmt.Sprintf("%v%v", prefix, "ClientBaseConfig.insecure"), defaultConfig.ClientBaseConfig.UseInsecureConnection, "Use insecure connection.")
+	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "ClientBaseConfig.maxBackoffDelay"), defaultConfig.ClientBaseConfig.MaxBackoffDelay.String(), "Max delay for grpc backoff")
+	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "ClientBaseConfig.perRetryTimeout"), defaultConfig.ClientBaseConfig.PerRetryTimeout.String(), "gRPC per retry timeout")
+	cmdFlags.Int(fmt.Sprintf("%v%v", prefix, "ClientBaseConfig.maxRetries"), defaultConfig.ClientBaseConfig.MaxRetries, "Max number of gRPC retries")
```

Comment:
This will be breaking change and all consumers of idl have to migrate to use the flags with new prefix 
Without this change the flag would have been 
--admin.endpoint
With the change it would be 
--admin.ClientBaseConfig.endpoint
which seems fine to make it work but doesn't seem to be a good UX.
Would have preferred that this refactoring wasn't going to change the api contract.

This is in line with my previous comment here https://github.com/flyteorg/flyteidl/pull/138#discussion_r631636458

The basic configuration of flytectl would have to change with consumption of this idl from this 
```
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns:///localhost:30081
  insecure: true
  authType: Pkce # if using authentication or just drop this. If insecure set insecure: True

```
to this
```
admin:
  ClientBaseConfig:
        # For GRPC endpoints you might want to use dns:///flyte.myexample.com
        endpoint: dns:///localhost:30081
        insecure: true
  authType: Pkce # if using authentication or just drop this. If insecure set insecure: True

```
https://docs.flyte.org/projects/flytectl/en/stable/#basic-configuration

Correct me if i am wrong here.

cc : @EngHabu @kumare3

Reply:
If that is the case, I can remove the refactoring for the [config](https://github.com/flyteorg/flyteidl/pull/138/commits/648256552b6759c6c4dd53dcd32b30e443ea1a39) which was trying address your [comment](https://github.com/flyteorg/flyteidl/pull/138#discussion_r630171297) so we can keep the backward compatibility. Let me know what do you think.

Reply:
Yeah we can remove the refactoring since it causing this problem. The code generator should possibly accept to not have this class name prefix by default in case of nested structures.